src/scripts/main.js

```diff
@@ -1,3 +1,231 @@
 'use strict';
 
-// write your code here
+const button = document.querySelector('.button');
+const score = document.querySelector('.game-score');
+
+const messageWin = document.querySelector('.message-win');
+const messageStart = document.querySelector('.message-start');
+const messageGameOver = document.querySelector('.message-lose');
+
+const fieldRow = document.querySelectorAll('.field-row');
+
+let totalScore = 0;
+const rows = 4;
+const columns = 4;
+
+const gameField = [
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+];
+
+function udateGameField() {
+  const randomRow = Math.floor(Math.random() * rows);
+  const randomCell = Math.floor(Math.random() * columns);
+  const randomNumber = Math.floor(Math.random() * 10);
+  const smallDigit = randomNumber < 9 ? 2 : 4;
+
+  if (gameField[randomRow][randomCell] === 0) {
+    gameField[randomRow][randomCell] = smallDigit;
+    updeteInfo();
+  } else {
+    udateGameField();
+    checkforfGameOver();
+  }
+
+  const arrOfNumbers = [];
+
+  for (const arr of gameField) {
+    for (const num of arr) {
+      arrOfNumbers.push(num)
+    }
+  }
+
+  let maxCellNum = Math.max(...arrOfNumbers);
+
+  if (maxCellNum === 2048) {
+    messageWin.classList.remove('hidden');
+  }
+
+};
+
+function updeteInfo() {
+  for (let row = 0; row < rows; row++) {
+    for (let cell = 0; cell < columns; cell++) {
+      if (gameField[row][cell] === 0) {
+        fieldRow[row].children[cell].textContent = '';
```

Comment:
Create a variable for ` fieldRow[row].children[cell]`. You use it not once

Reply:
Thank you.